src/java/org/apache/cassandra/repair/autorepair/AutoRepairUtils.java

```diff
@@ -600,18 +783,27 @@ else if (!hostIdsInCurrentRing.contains(nodeHistory.hostId))
                                 "First node in priority list is {}", ClusterMetadata.current().directory.addresses.get(NodeId.fromUUID(priorityHostId)));
                     return NOT_MY_TURN;
                 }
+
                 if (myId.equals(priorityHostId))
                 {
                     //I have a priority for repair hence its my turn now
                     return MY_TURN_DUE_TO_PRIORITY;
                 }
-                // get the longest unrepaired node from the nodes which are not running repair
-                AutoRepairHistory defaultNodeToBeRepaired = getHostWithLongestUnrepairTime(currentRepairStatus.historiesWithoutOnGoingRepair);
-                //check who is next, which is helpful for debugging
-                logger.info("Next node to be repaired for repair type {} by default: {}", repairType, defaultNodeToBeRepaired);
-                if (defaultNodeToBeRepaired != null && defaultNodeToBeRepaired.hostId.equals(myId))
+
+                // Determine if this node is the most eligible host to repair.
+                AutoRepairHistory nodeToBeRepaired = getMostEligibleHostToRepair(currentRepairStatus, myId);
```

Comment:
I've wavered between making whether or not to consider whether replicas are actively repairing should be configurable.  For now I've decided not to, as I can't see a downside of this other than that it might slow repair cycles down for smaller clusters where most nodes are replicas of each other.  Thoughts?